controllers/cluster_controller.go

```diff
@@ -417,6 +434,12 @@ func (c clusterDescendants) filterOwnedDescendants(cluster *clusterv1.Cluster) (
 		&c.workerMachines,
 		&c.controlPlaneMachines,
 	}
+	if feature.Gates.Enabled(feature.MachinePool) {
+		lists = append(lists, &c.machinePools)
+		copy(lists[4:], lists[3:])
+		lists[3] = &c.machinePools
+	}
```

Comment:
```suggestion
	if feature.Gates.Enabled(feature.MachinePool) {
		lists = append([]runtime.Object{&c.machinePools}, lists...)
	}
```

This _should_ work to prepend to the list